letsencrypt/client.py

```diff
@@ -237,12 +240,20 @@ def obtain_certificate(self, domains):
         :rtype: tuple
 
         """
+        authzr = self.auth_handler.get_authorizations(
+                domains,
+                self.config.allow_subset_of_names)
+
+        domains = [a.body.identifier.value.encode('ascii', 'ignore')
```

Comment:
I don't think `'ignore'` should be passed to `encode`. It should be impossible for this to fail because we encode every domain as `ascii` [here](https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/le_util.py#L313). If it does fail, it's a problem and we should error out.

Out of curiousity, did using `encode('ascii')`fail on you or were you just trying to lookout for errors in the future?